arch/risc-v/src/common/riscv_internal.h

```diff
@@ -279,6 +288,63 @@ int riscv_pause_handler(int irq, void *c, void *arg);
 
 uintptr_t riscv_mhartid(void);
 
+#ifdef CONFIG_ARCH_USE_S_MODE
+/* If kernel runs in Supervisor mode, declare proper function prototypes,
+ * this is because it is not possible to ecall from S mode to S mode
+ */
+
+int riscv_saveusercontext(uintptr_t *saveregs);
+void riscv_fullcontextrestore(uintptr_t *restoreregs) noreturn_function;
+void riscv_switchcontext(uintptr_t *saveregs, uintptr_t *restoreregs);
+void riscv_syscall_return(void);
+void riscv_syscall_dispatch(void) noreturn_function;
+
+#else
+
+#ifndef __ASSEMBLY__
```

Comment:
don't need, already check at line 30

Reply:
But we need avoid the redundant code if possible.

Reply:
Sure, I agree on that 100%. I'll remove the test here, just need to be careful if this file is modified or if these macros are moved elsewhere.